Update Docker scripts and README to DockerHub image names

mabelzhang opened the pull request:
I tested this. The commands in README should work with the new updates to the scripts.

Also added to README
- DockerHub URL
- `localhost:9090` added in #15 
- Troubleshoot section in case people want to use ROS 2 and Gazebo together
- Quick start section on main page

clalancette commented on docker/README.md:
````diff
@@ -2,32 +2,65 @@
+./run.bash osrf/icra2023_ros2_gz_tutorial:tutorial_no_nvidia --no-nvidia
+```
+
+If you built the image locally, this is a shorter, equivalent command:
 ```
 ./run.bash icra2023_tutorial
 ```
````
I'd honestly get rid of this.  The longer commands are fine (people are just going to cut-n-paste), and I think this can only lead to confusion.  But I'll leave it up to you.

mabelzhang replied:
Yeah... I thought about just having one. But this command is very convenient for someone who knows what's going on, because you can tab-complete the directory name and don't need to copy-paste. That's how I usually do it. Maybe I should add that.

The `osrf/` one is only there because that's the name of the image on DockedHub, and people who pull the image will need to use that. It's very long, and copy-pasting means people need to run `docker images` every time to copy it. Maybe we should add a line setting up an alias for quick access.

mabelzhang replied:
Added comments in in 97b343b

mabelzhang commented:
Addressed comments
I feature crept a bit and also added a Quick Start section to the main readme (multi-word anchors just need a `-` character between the words!), since it's not clear from the main page where people are supposed to go to get set up.
